src/ts/services/keyring-vault.ts

```diff
@@ -0,0 +1,171 @@
+import { KeyringInstance, KeyringPair, KeyringPair$Json } from '@polkadot/keyring/types'
+import Keyring from '@polkadot/keyring'
+import { Prefix } from '@polkadot/keyring/address/types'
+import { LocalStore } from './local-store'
+import { mnemonicGenerate, cryptoWaitReady, mnemonicValidate } from '@polkadot/util-crypto'
+import t from './i18n'
+
+class KeyringVault {
+
+  private _vaultKey: string = 'speckle-vault'
+  private _keyring?: KeyringInstance
+  private _password?: string
+  private _mnemonic?: string
+
+  private get keyring (): KeyringInstance {
+    if (this._keyring) {
+      return this._keyring
+    }
+    throw new Error(`Keyring is not initialised yet`)
+  }
+
+  unlock (password: string, addressPrefix?: Prefix): Promise<Array<KeyringPair$Json>> {
+    if (this.isUnlocked()) throw new Error(t('error.wallet.unlocked'))
+    if (!password.length) throw new Error(t('error.password'))
+    // this will be redundant if we have polkadot js api initialisation
+    return cryptoWaitReady().then(async () => {
+      this._keyring = new Keyring({ addressPrefix, type: 'sr25519' })
+      let vault = await LocalStore.get(this._vaultKey)
+      if (vault) {
+        let accounts = Object.values(vault[this._vaultKey])
```

Comment:
is local store key equal to account key in vault?

Reply:
yes, I think that's how LocalStore works. Correct me if I am wrong @mytechtip